0023-backend-v3.md

```diff
@@ -0,0 +1,239 @@
+
+The initial proposal in this RFC is to **maintain** the `BackendV2` design of the Target and extra functionality (up to small changes), and focus on improving the "run" component.
+
+### BackendV3 - the new backend model
+
+Instead of a required `backend.run()`, the proposal for `BackendV3` implementations is to offer **primitive factory methods**: `backend.sampler()` and `backend.estimator()` that return their respective instantiated primitive classes. These factory methods should be defined in the abstract `BackendV3` interface and it's up to the provider to decide what "flavor" of primitive will be returned in their specific `BackendV3` subclass. 
+
+The abstract definition of factory methods should be: 
+
+- as minimal as possible without being trivial
+- agnostic to the primitive interface version (to support both current and future versions of the primitives)
+
+We could additionally still maintain an optional `backend.run()` for 3rd party and local simulation backends.
+
+This sums up the change from having one mandatory implementation of the execution step (`.run()`) to having three optional implementations: 
+
+- `.sampler()`
+- `.estimator()`
+- `.run()`
+
+> A point that should be clarified is what we'd like the interface to require. With requiring `.run()` out of the question, should we make them all optional and not require backends to be executable? Should we require at least a `.sampler()` or a `.estimator()` method?
+
+It seems like not requiring an execution step is the most flexible alternative, as the need for a container that includes the target and the extra functionality still justify the need for having a backend object in these cases.
+
+-----
+Factors to consider:
+- Performance
+- Dependencies
+- Maintenance
+- Compatibility
+
+## Detailed Design
+
+This section traverses in more detail Let's traverse the different components of `BackendV2` how they would look like in `BackendV3`:
+
+#### 1. The Target (will not change in BackendV3)
+
+>`BackendV3` should still expose a `backend.target` attribute in the same fashion as `BackendV2`. This will return a `Target` instance.
+
+The `Target` class is a container for the different hardware specifications that model a QPU. It was modelled following the requirements of the Qiskit transpiler. It's designed to be vendor-agnostic, but allows for vendor-specific extensions.
+
+The `Target` defines 3 categories of information in its attributes:
+
+1. **Required:** necessary to define a fundamental instance. Qiskit (and in particular, the transpiler) relies on their definition. These are:
+    - number of qubits
+    - supported instructions, as tuples of `(instruction, qargs)`
+
+2. **Optional:** this information is not necessary, but if provided, Qiskit will offer tools to leverage this information and reason about it. These are:
+    - `QubitProperties` 
```

Comment:
One aspect that we got wrong in `BackendV2` is that `QubitProperties` started as a property of the `Backend` and not the `Target`. At the time the target was only thought of as what was needed for transpilation and we weren't consuming qubit properties in the transpiler. So we ended up in a weird state where it's optional in the backend and optional in the target to support people implementing `BackendV2` with qubit properties prior to qiskit-terra 0.20.0. I think in BackendV3 we should fix this and make it optional on the `Target` only and the `BackendV3` side alias a target query only.

Reply:
Added in [0427d3d](https://github.com/Qiskit/RFCs/pull/77/commits/0427d3da30df0bf05fcf405b92120e4cbfa4edc5)